### PR TITLE
Add language definition file for D

### LINKED ISSUE #5726: Add language definition file for D

### PR DESCRIPTION
**Pull request by amalrajan**
Closes #5726

### CODE REVIEW

**Comment by AkshJain99** [comment]
Your CI test failed you should make it green and moreover there is no description in commit message :wink:

File: coalib/bearlib/languages/definitions/D.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class D:
+    __qualname__ = 'D'
```
Comment (li-boxuan) [inline coalib/bearlib/languages/definitions/D.py]: This is unnecessary.

File: coalib/bearlib/languages/definitions/D.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    extensions = '.d'
```
Comment (li-boxuan) [inline coalib/bearlib/languages/definitions/D.py]: Needs to be a tuple.

**Comment by li-boxuan** [comment]
Need rebase.

File: coalib/bearlib/languages/definitions/D.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        'interface', 'invariant', 'ireal', 'is', 'lazy', 'long', 'macro',
+        'mixin', 'module', 'new', 'nothrow', 'null', 'out', 'override',
+        'package', 'pragma', 'private', 'protected', 'public', 'pure', 'real',
+        'ref', 'return', 'scope', 'shared', 'short', 'static', 'struct',
+        'super', 'switch', 'synchronized', 'template', 'this', 'throw', 'true',
+        'try', 'typedef', 'typeid', 'typeof', 'ubyte', 'ucent', 'uint',
+        'ulong', 'union', 'unittest', 'ushort', 'version', 'void', 'wchar',
+        'while', 'with', '__FILE__', '__FILE_FULL_PATH__', '__MODULE__',
+        '__LINE__', '__FUNCTION__', '__PRETTY_FUNCTION__', '__gshared',
+        '__traits', '__vector', '__parameters'
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/D.py]: please add a trailing comma

Comment (amalrajan) [reply coalib/bearlib/languages/definitions/D.py]: Thanks for reviewing. I think we should add a trailing comma to `keywords` inside other language definitions (example: C.py).

**Comment by subhaminion** [comment]
Test cases are failing. @amalrajan

**Comment by jayvdb** [comment]
Tests are fine.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5833/commits/4da9a1c8124878ea991503bd35ef0998de9eb498

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5833/commits/5b8610089f3f8daf23c1814828e033a4abe493f4

**Pull request merged**